From 408566ea433411898ad8d072371c09418d4b16be Mon Sep 17 00:00:00 2001
From: Pithlit <piglit@mail.de>
Date: Wed, 2 Nov 2022 15:55:04 +0100
Subject: [PATCH 7/7] set default values for auto stuff. Not sure if needed.

---
 src/spaceObjects/playerSpaceship.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/spaceObjects/playerSpaceship.h b/src/spaceObjects/playerSpaceship.h
index 4eff6876..4bbc60bf 100644
--- a/src/spaceObjects/playerSpaceship.h
+++ b/src/spaceObjects/playerSpaceship.h
@@ -92,9 +92,9 @@ public:
     // Time in seconds it takes to recalibrate shields
     float shield_calibration_delay;
     // Ship automation features, mostly for single-person ships like fighters
-    bool auto_repair_enabled;
-    bool auto_coolant_enabled;
-    bool auto_reload_tube_enabled;
+    bool auto_repair_enabled = false;
+    bool auto_coolant_enabled = false;
+    bool auto_reload_tube_enabled = false;
     // Whether shields are up (true) or down
     bool shields_active;
     // Password to join a ship. Default is empty.
-- 
2.39.1

